src/components/calcite-dropdown-item/calcite-dropdown-item.scss

```diff
@@ -55,13 +55,16 @@
   @apply ml-auto mr-0;
 }
 
+.calcite--rtl a.dropdown-link {
```

Comment:
It think you can make the selector less specific and remove the `a` by moving this to the `.calcite-rtl` selector below: https://github.com/Esri/calcite-components/blob/9cc831345bdf987eb14cd86ddb90bbb1d385ee2f/src/components/calcite-dropdown-item/calcite-dropdown-item.scss#L115-L120